pennylane/data/dataset.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the :class:`~pennylane.data.Dataset` class and its associated functions.
+"""
+
+from abc import ABC
+import dill
+import zstd
+
+
+class Dataset(ABC):
+    """The dataset data type. This class is used to create general datasets.
+    It accepts general keyword arguments to be flexible for different types of data.
+
+    Args:
+        dtype (string): Optional, the type of dataset, e.g. 'qchem' or 'qspin'
```

Comment:
I think it's better not to mention `dtype` as optional here.

Reply:
Edit made.